parser/src/parser.cpp

```diff
@@ -225,24 +250,62 @@ int main(int argc, char* argv[])
   if (vm.count("force"))
     cc::util::removeTables(db, SQL_DIR);
 
-  cc::util::createTables(db, SQL_DIR);
+  if (vm.count("force") || isNewDb)
+    cc::util::createTables(db, SQL_DIR);
 
   //--- Start parsers ---//
 
   cc::parser::SourceManager srcMgr(db);
   cc::parser::ParserContext ctx(db, srcMgr, compassRoot, vm);
   pHandler.createPlugins(ctx);
 
+  std::vector<std::string> reversedTopologicalOrder = pHandler.getTopologicalOrder();
+  std::reverse(reversedTopologicalOrder.begin(), reversedTopologicalOrder.end());
+
+  // TODO: Handle errors returned by preparse().
+  for(const std::string& parserName : reversedTopologicalOrder)
+  {
+    LOG(info) << "[" << parserName << "] preparse started!";
+    pHandler.getParser(parserName)->preparse();
+  }
+
+  // TODO: Consider whether there is a better place for this code.
+  (cc::util::OdbTransaction(ctx.db))([&]
+  {
+    for(auto& item : ctx.fileStatus)
+    {
+      switch (item.second)
+      {
+        case cc::parser::IncrementalStatus::MODIFIED:
+        case cc::parser::IncrementalStatus::DELETED:
+        {
+          LOG(info) << "Database cleanup: " << item.first;
+
+          // Fetch file from SourceManager by path
+          cc::model::FilePtr delFile = srcMgr.getFile(item.first);
+
+          // Delete File and FileContent (only when no other File references it)
+          srcMgr.removeFile(*delFile);
+          break;
+        }
+        case cc::parser::IncrementalStatus::ADDED:
+          // Empty deliberately
+          break;
+      }
+    }
+  });
+
   // TODO: Handle errors returned by parse().
   for (const std::string& parserName : pHandler.getTopologicalOrder())
   {
-    LOG(info) << "[" << parserName << "] started!";
+    LOG(info) << "[" << parserName << "] parse started!";
     pHandler.getParser(parserName)->parse();
   }
 
   //--- Add indexes to the database ---//
 
-  cc::util::createIndexes(db, SQL_DIR);
+  if(vm.count("force") || isNewDb)
```

Comment:
Space missing.

Reply:
Fixed in 4b52c300da8e3f8309356cf71e5f9c70207cc051.